### ISSUE TITLE
Lenses not being picked up correctly when used with Nikon Z series bodies.

### ISSUE THREAD

**Issue by ghost** [Beep6581/RawTherapee#6291]
First of all: I'm not sure how much of this is a lensfun issue and as such cannot be dealt with by the RawTherapee team. Just putting this out here just in case there is a RT part that can/needs to be fixed.

**Short description**
RawTherapee reports/uses the wrong lens info when using F-mount lenses on a Nikon Z6 II body.

**Expected behaviour**
Lenses to be picked up correctly by the _Profiled Lens Correction_ section.

**Additional information**
I have 4 Nikkor lenses, all of them are in the lesnfun database and they are picked up correctly by RT when using my Nikon D750 and incorrectly when using them on the Nikon Z6 II. (1 shows the wrong info in the Quick Info pop-up only).

When looking at RT's Lens Profile section I notice that the _Camera_ field is empty when choosing auto select. Reason being: The Nikon Z6 II isn't yet part of lensfun (see lensfun's [PR #1371](https://github.com/lensfun/lensfun/issues/1371)). It is present in RT's camconst.json file.

I thought that that might have been the problem, but manually adding this Z6 II info to lensfun's mil-nikon.xml doesn't solve the problem:

* The camera is picked up correctly after the manual add and the field is no longer empty.
*  The lens info is still wrong though.

**Screenshots**

Correct lens info would be:
![d75](https://user-images.githubusercontent.com/56407344/123507167-806c1d80-d657-11eb-9a36-dbba92accb31.png)

When using the same lens on the Z6 II:
![z62](https://user-images.githubusercontent.com/56407344/123507188-a265a000-d657-11eb-8e6f-e9e69f41eacf.png)

---
Before adding a lot of lens specific info, RAW examples and maybe other needed info I need to know if this is a RT issue or not.

If it is, please tell me which info I need to provide.

**Comment by ff2000** [comment]
Related issue:
https://github.com/Beep6581/RawTherapee/issues/5733

**Comment by ghost** [comment]
Some additional info:

If I look at the lens related information that exiv2 spits out I notice that there's more info for the Z6 when compared to the D750. Example:

D750 lens info info:

```
$ exiv2 -pa D75_105mm.nef | grep -i lens
Exif.Nikon3.LensType                         Byte        1  D G VR
Exif.Nikon3.Lens                             Rational    4  105mm F2.8
Exif.Nikon3.LensFStops                       Undefined   4  7
Exif.NikonLd3.LensIDNumber                   Byte        1  Nikon AF-S VR Micro-Nikkor 105mm f/2.8G IF-ED
Exif.NikonLd3.LensFStops                     Byte        1  F7.0
```

And this is from the Z6 with the same lens:

```
 $ exiv2 -pa Z62_105mm.nef | grep -i lens
Exif.Nikon3.LensType                         Byte        1  D G VR
Exif.Nikon3.Lens                             Rational    4  105mm F2.8
Exif.Nikon3.LensFStops                       Undefined   4  7
Exif.NikonLd4.LensIDNumber                   Byte        1  Nikon AF-S VR Micro-Nikkor 105mm f/2.8G IF-ED
Exif.NikonLd4.LensFStops                     Byte        1  F7.0
Exif.NikonLd4.LensID                         Short       1  0
Exif.Photo.LensSpecification                 Rational    4  1050/10 1050/10 280/100 280/100
Exif.Photo.LensMake                          Ascii       6  
Exif.Photo.LensModel                         Ascii      65  VR 105mm f/2.8G
Exif.Photo.LensSerialNumber                  Ascii      11  
```
This is true for all 4 lenses I can test this with.

I know that RT does not use exiv2 at the moment, but the way info is gathered might be influenced by the extra info that is present when using the Z6 II (I'm assuming this to be true for all the Nikon Z serie bodies).

If the above is at all important then here's a text file with the same info with all 4 lenses I have (50mm, 85mm, 105mm and 70-200mm):  [lens.info.txt](https://github.com/Beep6581/RawTherapee/files/6721152/lens.info.txt)

Here are example RAWs for the 105mm lens:

D750 + 105mm: https://fromsmash.com/D750-105mm-example (34Mb)
Z6II + 105mm: https://fromsmash.com/Z6II-105mm-example (29Mb)

Files are provided via *fromsmash* and the links expire on 2021/07/04. Let me know if it has and one still needs them (and/or examples for the other 3 lenses).

**Comment by Thanatomanic** [comment]
@jade-nl FWIW, I've downloaded the files. atm I don't have time to look into a fix...

**Comment by ghost** [comment]
@Thanatomanic : That's no problem for me; I've created dynamic rules to fix it for my Z6II+lenses. A general solution in due time would be nice.

**Comment by Lawrence37** [comment]
The problem is the full lens name is not being read from the metadata.
![image](https://user-images.githubusercontent.com/45837045/123557922-6fe7a000-d748-11eb-8485-9808686f96aa.png)

**Comment by kmilos** [comment]
If using exiv2, the updated logic for Z bodies (and D780 & D6) should be something like this:
```
if Exif.NikonLd4.LensID exist and is not 0:
  use standard Exif.Photo.LensModel (ASCII string that is ok for Z lenses and doesn't depend on infrequently updated exiv2 translation table)
elseif Exif.NikonLd4.LensID exist and is 0:
  FTZ adapter (or D780/D6), use translated Exif.NikonLd4.LensIDNumber
else:
  older D body, use the usual translated Exif.NikonLd3.LensIDNumber
```

**Comment by ghost** [comment]
This issue seems to be fixed once https://github.com/Beep6581/RawTherapee/pull/5889 is merged into dev. That PR is tagged for RT 6.0, though and not RT 5.10.
